### PR TITLE
Add GKE acceptance tests to Github Actions

### PR DESCRIPTION
**Pull request by BBBmau**
### Description

Implemented Acceptance Tests using GKE

### Acceptance tests
- [ ] Have you added an acceptance test for the functionality being added?
- [ ] Have you run the acceptance tests on this branch?

### Release Note
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-kubernetes/blob/main/CHANGELOG.md):

```release-note
* Add Acceptance Tests (GKE) to Github Actions
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

File: .github/workflows/acceptance_tests_gks.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      - name: Provision GKE Cluster
+        working-directory: ./kubernetes/test-infra/gke
+        env:
+          TF_VAR_cluster_version: ${{ github.event.inputs.clusterVersion }}
+          TF_VAR_workers_count: ${{ github.event.inputs.workersCount }}
+
+          TF_VAR_node_machine_type: ${{ github.event.inputs.nodeMachineType }}
+        run: |
+          terraform init
+          terraform apply -var=kubernetes_version=1.6.4
```
Comment (jrhouston) [inline .github/workflows/acceptance_tests_gks.yaml:69]: Don't need the `-var` here 😄

File: .github/workflows/acceptance_tests_gks.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
+name: Acceptance Tests (GKS)
```
Comment (jrhouston) [inline .github/workflows/acceptance_tests_gks.yaml:1]: Need to find and replace `GKS` -> `GKE`

File: .github/workflows/acceptance_tests_gks.yaml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+          role-duration-seconds: 14400
+      - name: Install Terraform
+        uses: hashicorp/setup-terraform@v1
+        with:
+          terraform_version: ${{ github.event.inputs.terraformVersion }}
+      - name: 'Set up Cloud SDK'
+        uses: 'google-github-actions/setup-gcloud@v0'
+      - name: 'Initialize gcloud SDK'
+        run: 'gcloud init'
+
```
Comment (jrhouston) [inline .github/workflows/acceptance_tests_gks.yaml:58]: Some random whitespace here that needs to be removed

File: .github/workflows/acceptance_tests_gks.yaml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+          path: ./.go-version
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ steps.go-version.outputs.content }}
+      - name: Authenticate to Google Cloud
+        uses: google-github-actions/setup-gcloud@v0
+        with:
+          credentials_json: '$GOOGLE_CREDENTIALS'
+          role-session-name: github-actions-tests-${{ github.run_number }}
```
Comment (jrhouston) [inline .github/workflows/acceptance_tests_gks.yaml:48]: I don't see this in the `setup-gcloud` action – I think you've copied this over from the EKS action by accident.

File: .github/workflows/acceptance_tests_gks.yaml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        id: go-version
+        uses: juliangruber/read-file-action@v1
+        with:
+          path: ./.go-version
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ steps.go-version.outputs.content }}
+      - name: Authenticate to Google Cloud
+        uses: google-github-actions/setup-gcloud@v0
```
Comment (jrhouston) [inline .github/workflows/acceptance_tests_gks.yaml:45]: I think you need to run their [auth](https://github.com/google-github-actions/auth#authenticating-via-service-account-key-json-1) action before you run the `setup-glcoud` action.

**Review by jrhouston — CHANGES_REQUESTED** [review]

File: kubernetes/test-infra/gke/main.tf [hunk 6]
```diff
@@ -3,11 +3,11 @@ variable "kubernetes_version" {
 }
 
 variable "workers_count" {
-  default = "1"
```
Comment (alexsomesan) [inline kubernetes/test-infra/gke/main.tf:6]: I would tweak this value via `-var` cmd line argument from a GH action script variable. I personally also use the configuration in `kubernetes/test-infra/gke` to spin up my own test envs for local development and I'd be nice to keep the size of the two independently adjustable.

**Review by jrhouston — APPROVED** [review]

**Pull request merged**